### PR TITLE
Enable support for generic methods with IDIC

### LINKED ISSUE #88964: IDynamicInterfaceCastable throws EntryPointNotFoundException when calling a generic method
### Description

When a type implementing `IDynamicInterfaceCastable` is used to obtain an interface with a generic method, calling this method ignores the `IDynamicInterfaceCastable.GetInterfaceImplementation` completely and instead throws (somewhat cryptic) `System.EntryPointNotFoundException`.

### Reproduction Steps

```cs
using System;
using System.Linq;
using System.Runtime.InteropServices;
					
public class Program
{
	public static void Main()
	{
		var castable = new Castable();
		Console.WriteLine("Querying interface");
		var test = (IQueryProvider)castable;
		Console.WriteLine("Calling interface");
		test.CreateQuery<int>(null!); // removing <int> works
	}
}

class Castable : IDynamicInterfaceCastable
{
	bool IDynamicInterfaceCastable.IsInterfaceImplemented(RuntimeTypeHandle interfaceType, bool throwIfNotImplemented)
	{
		Console.WriteLine("Checking interface " + Type.GetTypeFromHandle(interfaceType));
		return true;
	}

	RuntimeTypeHandle IDynamicInterfaceCastable.GetInterfaceImplementation(RuntimeTypeHandle interfaceType)
	{
		Console.WriteLine("Obtaining interface " + Type.GetTypeFromHandle(interfaceType));
		return default;
	}
}
```

### Expected behavior

The `GetInterfaceImplementation` implementation should be used (an exception in this case since a valid type is not returned anyway):
```
Querying interface
Checking interface System.Linq.IQueryProvider
Calling interface
Obtaining interface System.Linq.IQueryProvider
Unhandled exception. System.InvalidCastException: Invalid cast from 'ImplementsAnything' to 'System.Linq.IQueryProvider'.
   at System.Linq.IQueryProvider.CreateQuery(Expression expression)
   at Program.Main()
```

### Actual behavior

`GetInterfaceImplementation` is ignored completely:
```
Querying interface
Checking interface System.Linq.IQueryProvider
Calling interface
Unhandled exception. System.EntryPointNotFoundException: Entry point was not found.
   at Program.Main()
```

### Regression?

Does not work in any version of .NET from 5 to 8.

### Known Workarounds

_No response_

### Configuration

.NET 8 Preview 3

### Other information

_No response_

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #88964

Create tests to validate scenario.

Support for NativeAOT is still blocked - see https://github.com/dotnet/runtime/issues/72909.

### CODE REVIEW

File: src/tests/Interop/IDynamicInterfaceCastable/Program.cs [hunk 1]
```diff
@@ -373,6 +404,8 @@ private static void ValidateGenericInterface()
             Assert.Equal(expectedInt, testInt.ReturnArg(42));
             Assert.Equal(expectedStr, testStr.ReturnArg(expectedStr));
             Assert.Equal(expectedStr, testVar.ReturnArg(expectedStr));
+            Assert.Equal(expectedInt * 2, testInt.DoubleGenericArg<int>(42));
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/tests/Interop/IDynamicInterfaceCastable/Program.cs:407]: I have the `DoubleGenericArg<string>(...)` below. Is that not sufficient? Happy to add more options if desired though.

File: src/tests/Interop/IDynamicInterfaceCastable/Program.cs [hunk 2]
```diff
@@ -149,6 +165,21 @@ public interface ITestGenericIntImpl: ITestGeneric<int, int>
...
+
+        V ITestGeneric<int, int>.DoubleGenericArg<V>(V v)
+        {
+            if (v is int i)
+            {
+                i *= 2;
+                return Unsafe.As<int, V>(ref i);
+            }
+            else if (v is string s)
+            {
```
Comment (jkotas) [inline src/tests/Interop/IDynamicInterfaceCastable/Program.cs:177]: Can you also validate that `typeof(V) == typeof(string)` here?It  will make ensure that the hidden generic arg is correct.

File: src/tests/Interop/IDynamicInterfaceCastable/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkotas) [reply src/tests/Interop/IDynamicInterfaceCastable/Program.cs:407]: Ah ok, I have missed the second test here.

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**